CREST addition to bde workflow

shreesowndarya opened the pull request:
I have done it in a new file because the way we write gaussian gif file changes from mol object to .xyz from CREST. I wasn't sure if this would become the standard procedure so I did it as a separate file.

Things done:

1. Adding in crest after MMFF optimization (mol object is converted to .xyz following which CREST is done)
2. based on best conformers (xyz) from CREST, the gjf file is written now from the crest_best.xyz file

I debugged independently the following:

1. mol object after MMFF optimization to xyz file for CREST
2. writing of .gjf file from the crest_best.xyz 

Checks needed:
How crest is invoked on eagle and if I set up the temporary directories correctly!

Once the .gjf file is written the protocol is the same afterwards.

shreesowndarya committed:
1. Adding in crest after MMFF optimization
2. based on best conformers (xyz), the gjf file is written now from the 
xyz file
3. Only check : How crest is invoked on eagle

pstjohn commented on bde/gaussian_crest.py:
```diff
@@ -0,0 +1,282 @@
+        rdkit.Chem.rdmolfiles.MolToXYZFile(mol,self.xyz,confId=confId)
+
+        #running calc in temporary TemporaryDirectory : NEEDS CHECK
+        with tempfile.TemporaryDirectory(dir=tmpdirname) as cresttmp:
```
I'm not sure this is actually doing anything? you could put the other output in cresttmp, or chdir into cresttmp, but this is all already happening inside a temporary directory (tmpdirname) so the nested temp directory (i.e., what I did for gaussian) is probably not necessary

pstjohn commented on bde/gaussian_crest.py:
```diff
@@ -0,0 +1,282 @@
+            subprocess.run(crest_cmd, shell=True, env=env,
+                           timeout=self.crest_timeout)
+
+            self.best_xyz = tmpdirname + '/{0}_{1}_best.xyz.'.format(self.cid, self.run_hex)
```
this could probably get moved with the other path variables in lines 133-135

pstjohn commented:
looks great! the install for CREST does seem a bit onerous, even if xtb is installable with conda
https://github.com/grimme-lab/crest

I wonder if their intel binaries would work on eagle

shreesowndarya committed:
Addressing comments on pull request

1. Removed nested temp directory for crest
2. moved with the other path variables in lines 133-135

shreesowndarya commented:
Should I contact HPC about how xtb/crest can be installed on eagle??

shreesowndarya committed:
1. updating title line for gjf files
2. debugging for both molecule and fragment
